src/emit/binary.rs

```diff
@@ -189,11 +186,12 @@ impl<'a> Binary<'a> {
         context: &'a Context,
         target: Target,
         name: &str,
-        filename: &str,
+        filename: String,
```

Comment:
Nitpick: Why was this changed? You use `filename.as_str()` later on anyways. In general, if you do not need ownership over a value, it is better to ask for a reference.